### ISSUE TITLE
[JIT] support self.named_buffers and self.named_parameters in TorchScript

### ISSUE THREAD

**Issue by chenbohua3** [pytorch/pytorch#36211]
## 🚀 Feature
 support self.named_buffers and self.named_parameters in TorchScript

## Motivation

Since `self.named_models` had been supported in torchscript [#29495 ]( https://github.com/pytorch/pytorch/pull/29495), it will be more convenient to support self.named_buffers and self.named_parameters. Projects like object detection and model compression use buffers to stores important things (e.g. anchor box and pruning mask).

## Pitch
```
  class MyMod(torch.nn.Module):
      def __init__(self):
          super(MyMod, self).__init__()
          self.register_buffer('x', torch.zeros(1))

      def forward(self, x):
          for name, buffer in self.named_buffers()
              x = x + buffer
          return x

  model = MyMod()
  torch.jit.script(model)
```

**Comment by chenbohua3** [comment]
@eellison :)

**Comment by voznesenskym** [comment]
Alright @chenbohua3 the diffs above should fix things for you. As a good sanity check, do you mind pulling them down and confirming?

**Comment by chenbohua3** [comment]
@voznesenskym Sure. thanks for your effort, I will confirm this diff:)

**Comment by chenbohua3** [comment]
@voznesenskym Sorry for the late reply. I have tested ur diff (71481b4245f0618773b6a389d4a8d21a847d5963) on Detectron2 and It works well:)

**Comment by voznesenskym** [comment]
Perfect @chenbohua3! (cc: @suo) I will work on getting the second diff here landed ASAP for you.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
